src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinRTWebSocket.cs

```diff
@@ -105,6 +114,21 @@ public override string SubProtocol
                 websocketControl.SupportedProtocols.Add(subProtocol);
             }
 
+            if (options.ClientCertificates.Count > 0)
+            {
+                if (!MessageWebSocketClientCertificateSupported)
+                {
+                    throw new NotSupportedException(string.Format(
```

Comment:
We usually throw PlatformNotSupportedException for things like this.  See recent HttpClient PR.

The NotSupportedException is ok for the other case about certain client certificates not working.  But for this exception, it should be PlatformNotSupportedException because the user can solve this by upgrading the platform to a more recent Windows 10 OS.